### DISCUSSION TITLE
Get template details in terraform

### DISCUSSION THREAD

**Discussion by samkeet** [coder/coder#13122] [General] [answered]
I am running an older version of Coder - 2.0.2. As a part of the terraform script, I wanted to fetch the terraform template details as per the [terrform guide](https://registry.terraform.io/providers/coder/coder/latest/docs/data-sources/workspace#template_version). 

How ever these values are all empty. 
```
"${data.coder_workspace.me.template_id}"
"${data.coder_workspace.me.template_name}"
"${data.coder_workspace.me.template_version}"
```

Are they tied to a coder version?

**Comment by matifali** [comment, marked as answer]
The best way to test this is to use the latest version. 2.10.2. Could you test and report back?


It would help if Terraform provider docs were updated to show the minimum Coder version required for a feature to work. 
cc: @mtojek @stirby

**Reply by ammario** [reply]
https://github.com/coder/terraform-provider-coder/issues/224 tracks the broader problem of improving the version mismatch experience.

**Comment by samkeet** [comment]
> The best way to test this is to use the latest version. 2.10.2. Could you test and report back?
> 
> 
> 
> 
> 
> It would help if Terraform provider docs were updated to show the minimum Coder version required for a feature to work. 
> 
> cc: @mtojek @stirby 

Confirmed that it's related to our coder version! I validated with 2.5.1.

I agree that Terraform docs should reflect a minimum coder version

Thank you!

**Comment by mtojek** [comment]
As https://github.com/coder/terraform-provider-coder/issues/224 is merged now, let's close this discussion.
